### ISSUE TITLE
[suggestion] add transformer name / embedding names in the model name 

### ISSUE THREAD

**Issue by lfoppiano** [kermitt2/delft#130]
I'm thinking that could be useful to have an additional information in the model name, such as the transformer name or the embeddings name. We could implement a sort of name compression (e.g. removing the slashes in the huggingface names to make it more readable). 

The motivation would be to allow to cohesist of models trained with different transformers/embeddings 

What do you think?

**Comment by kermitt2** [comment]
It was my first idea, I added very early the embeddings name in the model name, then removed it. 

The problem is that we can have several embeddings, and in the future several transformers in the same architecture. They all can be combined so it will be unreadable and it will be hard to have canonical names with so many pieces. 
Daniel also raised the point that there was too much "magic" in the way the path to a model was solved (using its name) and we should avoid using the name to "guess" attributes of the model - which is a very good point, and there are still a few places where the name is used (I try to remove it). 

On the other hand, the library supports arbitrary name, there is only managed in the `applications/` part, so it does not prevent a user of the library to change that.
